test/integration/deployment/deployment_test.go

```diff
@@ -498,3 +498,108 @@ func TestRollbackDeploymentRSNoRevision(t *testing.T) {
+			t.Fatalf("failed to update deployment %q to revision 1: %v", dname, err)
+		}
+		// Make sure the deployment to complete while manually marking its pods as ready at the same time
+		if err = tester.waitForDeploymentCompleteAndMarkPodsReady(); err != nil {
+			t.Fatalf("deployment %q failed to complete: %v", dname, err)
+		}
+	}
+
+	// Both deployments should proceed independently
+	rsSelector, err := metav1.LabelSelectorAsSelector(testers[0].deployment.Spec.Selector)
+	if err != nil {
+		t.Fatalf("failed to parse deployment selector: %v", err)
+	}
+	rsList, err := c.ExtensionsV1beta1().ReplicaSets(ns.Name).List(metav1.ListOptions{LabelSelector: rsSelector.String()})
+	if err != nil {
+		t.Fatalf("failed to list replicasets of deployments %q and %q in namespace %q: %v", firstDeploymentName, secondDeploymentName, ns.Name, err)
+	}
+	rss := rsList.Items
+	if len(rss) != 2 {
+		t.Fatalf("expected 2 replicasets to be found, but found %d", len(rss))
+	}
+
+	firstController := metav1.GetControllerOf(&rss[0])
+	if firstController == nil {
+		t.Fatalf("replicaset %q is not owned by any deployment", rss[0].Name)
+	}
+	secondController := metav1.GetControllerOf(&rss[1])
+	if secondController == nil {
+		t.Fatalf("replicaset %q is not owned by any deployment", rss[1].Name)
+	}
+
+	firstDeploymentUID := testers[0].deployment.UID
+	secondDeploymentUID := testers[1].deployment.UID
+	if !((firstController.UID == firstDeploymentUID && secondController.UID == secondDeploymentUID) ||
+		(firstController.UID == secondDeploymentUID && secondController.UID == firstDeploymentUID)) {
+		t.Fatalf("expected replicasets %q and %q are owned by deployments %q and %q, but found deployments %q and %q", rss[0].Name, rss[1].Name, firstDeploymentName, secondDeploymentName, firstController.Name, secondController.Name)
+	}
+
+	// Scale only the first deployment
+	newReplicas := int32(2)
+	testers[0].deployment, err = testers[0].updateDeployment(func(update *v1beta1.Deployment) {
+		update.Spec.Replicas = &newReplicas
+	})
+	if err != nil {
+		t.Fatalf("failed updating deployment %q: %v", firstDeploymentName, err)
+	}
+
+	// Wait for the controller to notice the scale
+	if err := testers[0].waitForObservedDeployment(testers[0].deployment.Generation); err != nil {
+		t.Fatalf("encountered an error when controller waits to notice the scale: %v", err)
+	}
+
+	// Verify that only replicaset of first deployment is scaled
+	rsList, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).List(metav1.ListOptions{LabelSelector: rsSelector.String()})
+	if err != nil {
+		t.Fatalf("failed to list replicasets of deployments %q and %q in namespace %q: %v", firstDeploymentName, secondDeploymentName, ns.Name, err)
+	}
+	rss = rsList.Items
+	if len(rss) != 2 {
+		t.Fatalf("expected 2 replicasets to be found, but found %d", len(rss))
+	}
+	if !((*rss[0].Spec.Replicas == newReplicas && *rss[1].Spec.Replicas == replicas) ||
+		(*rss[0].Spec.Replicas == replicas && *rss[1].Spec.Replicas == newReplicas)) {
```

Comment:
Need to make sure the right RS is scaled. In this check, we aren't sure which RS belongs to the deployment being scaled. 

It'd be easier to use a map to store the [controllerRef: rs] mapping. You can use it to compare deployments UIDs with controllerRef, and to compare rs replicas with deployment.

Reply:
After careful thinking, this step may be unnecessary. One of the previous steps verified the two replicasets have intended two deployments as their controllers (e.g., first replicaset has first deployment as its controller, and vice versa).

After scaling, we only update the first deployment. Only first deployment performs rollout, and thus only its replicaset will be scaled. Correct me if I am wrong.

Reply:
The purpose of this test is to make sure that overlapping deployment won't interfere with each others, so this check makes sure that the first deployment's scale won't affect any replicaset but its own.